consolidate spark images

mattf opened the pull request:
remove the master & worker images. their contents is merged into the base image.

the base should be started with /spark-master to get the master and /spark-worker to get the worker.

additional updates are needed to the kubernetes/example/spark/*-controller.yaml files. i'll follow this with the appropriate pr

Commit:
make all log4j.properties the same

Commit:
use only the base log4j.properties

Commit:
remove unnecessary EXPOSE

Commit:
merge worker into base image

Commit:
merge master into base image

zmerlynn commented on spark/Makefile:
```diff
@@ -17,10 +17,6 @@ ZEPPELIN_TAG = v0.5.5_v3
-	docker build -t gcr.io/google_containers/spark-master master
-	docker tag gcr.io/google_containers/spark-master gcr.io/google_containers/spark-master:$(TAG)
 	docker build -t gcr.io/google_containers/spark-driver driver
 	docker tag gcr.io/google_containers/spark-driver gcr.io/google_containers/spark-driver:$(TAG)
```
Why keep the driver image? It's barely an entrypoint pivot off the other two.

zmerlynn replied:
(i.e. I *think* we can just turn the daemons into different commands called from the -controller.yaml files and the driver script can just be the default image.)

mattf replied:
i could go either way on the driver. my decision to leave it was essentially deferring its fate. it would be nice if the driver setup the environment no matter what command is run.

mattf replied:
once this change is in i have a pr for the kubernetes repository proper that includes things like -

\-          image: gcr.io/google_containers/spark-master:1.5.1_v2
\+          image: gcr.io/google_containers/spark-base:1.5.1_v2
\+          command: ["/start-master"]

zmerlynn replied:
I think it would actually just be easier to consolidate this to `google_containers/spark:1.5.1_v3` and merge them all. Up for that?

Commit:
merge driver into base, aka remove driver

the base image does not have a cmd or entrypoint, making it usable for
a driver container

Commit:
move base image into top level directory, it's now just the spark image

Commit:
update version bump instructions

Commit:
bump version to 1.5.1_v3

mattf commented:
@zmerlynn i've removed the driver, pulled base/* up a level and bumped to v3

Commit:
fix incorrect tagging, base no longer exists

zmerlynn commented:
This LGTM. We'll definitely need to update the example. Sorry for the slooooow reply.

zmerlynn commented:
And this is being pushed as `gcr.io/google_containers/spark:1.5.1_v3` now.
